### PR TITLE
feat(nlu-client): allow nlu client extension by setting attributes to protected

### PR DESCRIPTION
**Pull request by michaelmass**

### CODE REVIEW

File: packages/nlu-client/src/oauth.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const createOauthTokenClient = (axios: AxiosInstance, { url, client_id, client_secret }: OauthTokenClientProps) => () =>
+  axios({
+    url,
+    method: 'post',
+    data: qs.stringify({ client_id, client_secret, grant_type: 'client_credentials' })
+  }).then((res) => res.data)
+
+const getMaxAge: any = (res: any) => res.expires_in * 1000
+const headerFormatter = (res: any) => 'bearer ' + res.access_token
```
Comment (spg) [inline packages/nlu-client/src/oauth.ts:19]: Maybe prefer template strings (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

**Review by spg — APPROVED** [review]

File: packages/nlu-client/src/oauth.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+const getMaxAge: any = (res: any) => res.expires_in * 1000
+const headerFormatter = (res: any) => `bearer ${res.access_token}`
+
+const interceptor = (provider: typeof tokenProvider, authenticate: () => Promise<string>) => {
+  const getToken = provider.tokenCache(authenticate, { getMaxAge })
+  return provider({ getToken, headerFormatter })
+}
+
+interface OauthClientProps {
```
[the remark this answers is not in the document]
Comment (EFF) [reply packages/nlu-client/src/oauth.ts:26]: We also tend to declare types and interface at the top of files as it's much easier to refac when it's time to extract types in dedicated typing files

**Review by EFF — CHANGES_REQUESTED** [review]
Few code standard to fix, other than that, logic is LGTM

**Review by franklevasseur — APPROVED** [review]
You'll need to increase the version of `nlu-client` by a patch so I can republish it on npm

**Pull request merged**